value/userguide/index.md

```diff
@@ -74,21 +74,31 @@ examples short and simple.
 Maven users should add the following to the project's `pom.xml` file:
 
 ```xml
+<dependency>
+  <groupId>com.google.auto.value</groupId>
+  <artifactId>auto-value-annotations</artifactId>
```

Comment:
unless the annotations are now runtime retention... is it really necessary to add this as a strict dep to maven projects?

Reply:
The annotations (at least most of them) now have `CLASS` retention, so it isn't completely pointless to have them at runtime. I suppose we could tell people that they can give this dependency `provided` scope if they want, but is it worthwhile for such a tiny dependency?

Reply:
FYI: Guava (and, I recently learned from jbduncan, [JUnit 5](https://github.com/junit-team/junit5/issues/1105)) have stopped using `provided` scope for annotation dependencies after [various](https://github.com/google/guava/issues/1018) [discussions](https://github.com/google/guava/issues/2721) [of](https://github.com/google/guava/issues/2824) [problems](https://github.com/google/guava/issues/2837). In short, `provided` [doesn't mean "compile-time"](https://issues.apache.org/jira/browse/MNG-2205?focusedCommentId=16267311&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-16267311) as you'd probably hope; instead, it means "compile-time for this project and missing entirely in downstream compilations." This causes problems for people who enable all `javac` warnings or use some annotation processors.

Reply:
We should probably add a note in `index.md` suggesting that people *not* use `provided` scope, then, with your [link](https://issues.apache.org/jira/browse/MNG-2205?focusedCommentId=16267311&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-16267311).

Reply:
Hmm, then the Gradle snippet should probably use `api` (for [`java-library`](https://docs.gradle.org/current/userguide/java_library_plugin.html#sec:java_library_configurations_graph) and [Android](https://developer.android.com/studio/build/gradle-plugin-3-0-0-migration.html#new_configurations) projects) or `compile` (for all others).

Though this actually depends whether your autovalues are part of your public API or not.

BTW, `apt` should be updated to `annotationProcessor`, and with Gradle 4.6 the `net.ltgt.apt` plugin is no longer required (though it still provides useful features).

So how about:

> Gradle users should use the following in their `build.gradle` script:
> 
> ```groovy
> dependencies {
>   // Use 'api' rather than 'compile' for Android or java-library projects.
>   compile             "com.google.auto.value:auto-value-annotations:1.6"
>   annotationProcessor "com.google.auto.value:auto-value:1.6"
> }
> ```
> You may want, or have to, install the [`net.ltgt.apt`][tbroyer-apt] plugin in non-Android projects.
> 
> [tbroyer-apt]: https://github.com/tbroyer/gradle-apt-plugin

Note that I updated the link to my plugin to point to the GitHub repo where the README provides documentation, including about companion plugins for better IDE support, whereas the Gradle Plugin Portal only has code snippets (I've had a couple reports already from people who hadn't seen the companion plugins, coming from the Plugin Portal).

And in the note you suggest to add about `<scope>provided</scope>` (or `<optional>true</optional>`), add a note about the `compileOnly` and `implementation` Gradle configurations (that's the technical name in Gradle, "configurations", not "scopes").

Reply:
Hm, shouldn't be `compileOnly` recommended for Gradle users then? 

This seems like best configuration: annotations will be available to javac and annotation processors (including 3rd-party "extension" processors) of the module, but won't be included into jar nor will be transitively available to other modules.

Afaik philosophy of Auto* was to be hidden from runtime introspection by default https://github.com/google/auto/issues/290#issuecomment-328547302

Reply:
@artem-zinnatullin See comments above, specifically https://github.com/google/auto/pull/606#discussion_r177278685 and https://github.com/google/auto/pull/606#discussion_r177441884. The issue is about being present at compile-time (or static analysis time) of _downstream_ projects.